src/api/routes/mentorBookings.ts

```diff
@@ -30,6 +32,31 @@ const create = procedure
       topic,
     }, { transaction });
 
+    // If a specific mentor was requested, create a chat message
+    if (requestedMentorId) {
+      try {
+        // Get the requested mentor user object
+        const requestedMentor = await db.User.findByPk(requestedMentorId, {
+          attributes: ['id', 'name', 'roles'],
+          transaction
+        });
+
+        if (requestedMentor) {
+          const room = await findOrCreateRoom(requestedMentor, me.id, transaction);
+          const markdown = `【不定期导师预约】学生预约话题：${topic}`;
+          await createMessageAndScheduleEmail(
+            requestedMentor,
```

Comment:
the message author should be the mentee who creates the request and not the mentor.